sdk/ai/azure-ai-agents/samples/agents_async/sample_agents_deep_research_async.py

```diff
@@ -42,51 +43,171 @@
 
+    # Write progress to file
+    with open(progress_filename, "a", encoding="utf-8") as fp:
+        fp.write("\nAGENT>\n")
+        fp.write(agent_text)
+        fp.write("\n")
+
+        for ann in response.url_citation_annotations:
+            fp.write(f"Citation: [{ann.url_citation.title}]({ann.url_citation.url})\n")
+
     return response.id
 
 
-def create_research_summary(message: ThreadMessage, filepath: str = "research_summary.md") -> None:
+def create_research_summary(message: ThreadMessage, filepath: str = "research_report.md") -> None:
+    """
+    Create a formatted research report from an agent's thread message with numbered citations
+    and a references section.
+
+    Args:
+        message (ThreadMessage): The thread message containing the agent's research response
+        filepath (str, optional): Path where the research summary will be saved.
+            Defaults to "research_report.md".
+
+    Returns:
+        None: This function doesn't return a value, it writes to a file
+    """
     if not message:
-        print("No message content provided, cannot create research summary.")
+        print("No message content provided, cannot create research report.")
         return
 
     with open(filepath, "w", encoding="utf-8") as fp:
         # Write text summary
         text_summary = "\n\n".join([t.text.value.strip() for t in message.text_messages])
+        # Convert citations to superscript format
+        text_summary = convert_citations_to_superscript(text_summary)
         fp.write(text_summary)
 
-        # Write unique URL citations, if present
+        # Write unique URL citations with numbered bullets, if present
         if message.url_citation_annotations:
-            fp.write("\n\n## References\n")
+            fp.write("\n\n## Citations\n")
             seen_urls = set()
+            citation_dict = {}
+
             for ann in message.url_citation_annotations:
                 url = ann.url_citation.url
                 title = ann.url_citation.title or url
+
                 if url not in seen_urls:
-                    fp.write(f"- [{title}]({url})\n")
+                    # Extract citation number from annotation text like "【58:1†...】"
+                    citation_number = None
+                    if ann.text and ":" in ann.text:
+                        match = re.search(r"【\d+:(\d+)", ann.text)
+                        if match:
+                            citation_number = int(match.group(1))
+
+                    if citation_number is not None:
+                        # Only add if this citation number isn't already used
+                        if citation_number not in citation_dict:
+                            citation_dict[citation_number] = f"[{title}]({url})"
+                    else:
+                        # Fallback for citations without proper format
+                        fallback_num = len(citation_dict) + 1
```

Comment:
In this case if we will get citations `58:%number%†` and  citations without number, we can overwrite the "official" citate by the fallback one and vice versa. I think the best approach would be to have two a dictionary with citations ordinals, based on the whole content of a citate number:number†source -> ordinal and keep the list of citates-urls, so that we can take the user-friendly link as text_citation_list[citations["number:number†source"]]